controller/transfer/record_transfer_manager.py

```diff
@@ -63,11 +63,10 @@ def import_records_and_rlas(
             labels_data,
             tasks_data,
         ) = split_record_data_and_label_data(chunk)
-
-        primary_keys = None
+        
         if idx == 0:
             create_attributes_and_get_text_attributes(project_id, records_data)
-            primary_keys = attribute.get_primary_keys(project_id)
+        primary_keys = attribute.get_primary_keys(project_id)
```

Comment:
they shouldn't change within the loop so only requesting them once is valid imo. The initialization to None is faulty (or needs to be done outside the loop)

Reply:
You are right, this is about primary keys and they should not change during the transaction even if the user triggers a change during the execution by the frontend. Will change this back!